src/ConstructiveSolidGeometry/SurfacePrimitives/Polygon.jl

```diff
@@ -0,0 +1,177 @@
+
+struct Polygon{N,T} <: AbstractPlane{T}
+    points::SVector{N, CartesianPoint{T}} 
+end
+const Triangle{T} = Polygon{3, T}
+const Quadrangle{T} = Polygon{4, T}
+
+normal(p::Polygon) = normalize((p.points[2] - p.points[1]) × (p.points[3] - p.points[1]))
+
+vertices(p::Polygon) = p.points
+
+
+function edges(p::Triangle{T}) where {T}
+    vs = vertices(p)
+    return SVector{3,Edge{T}}(
+        Edge(vs[1], vs[2]),
+        Edge(vs[2], vs[3]),
+        Edge(vs[3], vs[1])
+    )
+end
+function edges(p::Quadrangle{T}) where {T}
+    vs = vertices(p)
+    return SVector{4,Edge{T}}(
+        Edge(vs[1], vs[2]),
+        Edge(vs[2], vs[3]),
+        Edge(vs[3], vs[4]),
+        Edge(vs[4], vs[1])
+    )
+end
+
+Plane(p::Polygon{N, T}) where {N, T} = Plane{T}(p.points[1], (p.points[2] - p.points[1]) × (p.points[3] - p.points[1]))
+
+function _get_rot_for_rotation_on_xy_plane(p::Polygon{<:Any, T}) where {T}
+    n = normal(p)
+    rot_angle_rad = CartesianVector{T}(0,0,1) ⋅ n
+    return if abs(rot_angle_rad) == 1
+        RotationVec(zero(T), zero(T), zero(T))
+    else
+        rot_angle = -acos(rot_angle_rad)
+        rot_axis = rot_angle * normalize(CartesianVector{T}(0,0,1) × n)
+        RotationVec(rot_axis[1], rot_axis[2], rot_axis[3])
+    end
+end
+
+function _rotate_on_xy_plane(p::Polygon{<:Any, T}) where {T}
+    rot = _get_rot_for_rotation_on_xy_plane(p)
+    map(pt -> rot * pt, p.points)
```

Comment:
sometimes `broadcast` is faster than `map`. Have you checked this?

Reply:
I just tested it and here it does not make a difference.